template/src/features/counter/counterSlice.js

```diff
@@ -1,4 +1,20 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+import { fakeAsyncCall } from './fakeApi';
+
+// The function produced by `createAsyncThunk` below is called a thunk.
+// It allows us to perform async logic. It can be dispatched like a regular action:
+// `dispatch(incrementAsync(10))`.
+// The `thunkApi` argument has `dispatch`, `getState` and other functions available.
+// See https://redux-toolkit.js.org/api/createAsyncThunk
+// Async code can then be executed and other actions can be dispatched.
+export const incrementAsync = createAsyncThunk(
+  'counter/incrementAsync',
+  async (amount, thunkApi) => {
+    const value = await fakeAsyncCall(amount);
+    return value;
+  }
```

Comment:
The result of `fakeAsyncCall` could just be returned directly if it's already a Promise.

```suggestion
 (amount, thunkApi) => fakeAsyncCall(amount)
```